### ISSUE TITLE
'Nonce does not match' error when state data contains '=' encoded as %3D

### ISSUE THREAD

**Issue by DigitalFoundry-Brandon** [auth0/auth0.js#377]
This issue occurs when btoa appends an '=' to the stringfy data object for base64 encoding. The state data then contains a %3D encoded character after it round trips from IDP login which causes TransactionManager to attempt to locate the transaction as 'ASDF%3D' instead of 'ASDF='

```
	    this.auth0.authorize({
			connection: authProviders.google,
			redirectUri: document.location.protocol + "//" + document.location.host + "/register",
			state: 'ASDF='
		})
}

```
Issue can be resolved by adding the following on line 51 of transaction-manager.js
```
transaction = transaction.replace(/%3D/ig, '=')
```

Occurs in chrome v56

**Comment by DigitalFoundry-Brandon** [comment]
I've updated my own code to strip the padding character. Not sure this should be fixed in auth0 as = should not be passed in the first place.

**Comment by hzalaz** [comment]
We should handle the url decoding since the state comparison is done by auth0.js.

**Comment by hzalaz** [comment]
Closed by #382
